packages/fluentui/react-northstar/src/components/Table/Table.tsx

```diff
@@ -90,43 +92,33 @@ export const Table: React.FC<WithAsProp<TableProps>> &
   });
 
   const renderRows = () => {
-    return _.map(rows, (row: TableRowProps, index: number) => {
-      const props = {
-        compact,
-        onClick: (e, props) => {
-          _.invoke(row, 'onClick', e, props);
-        },
-      } as TableRowProps;
-      const overrideProps = handleVariablesOverrides(variables);
+    return _.map(rows, (row: TableRowProps) => {
       return TableRow.create(row, {
         defaultProps: () =>
           getA11yProps('row', {
-            ...props,
+            compact,
+            onClick: (e, props) => {
+              _.invoke(row, 'onClick', e, props);
+            },
           }),
-        overrideProps,
+        overrideProps: predefinedProps => ({
+          variables: mergeVariablesOverrides(variables, predefinedProps.variables),
+        }),
       });
     });
   };
 
   const renderHeader = () => {
-    if (!header) {
```

Comment:
Should we really remove this check?

Reply:
Yup, as `.create()` call few lines below handles this:

https://github.com/microsoft/fluentui/blob/e5adec2a91f1b920d1b0a2ab5e29828f2a7f4c30/packages/fluentui/react-northstar/src/components/Table/Table.tsx#L111-L124